bson/src/test/unit/org/bson/json/UuidStringValidatorTest.java

```diff
@@ -0,0 +1,107 @@
+            "11111111-1111-1111-1111-111111111111",
+            "22222222-2222-2222-2222-222222222222",
+            "33333333-3333-3333-3333-333333333333",
+            "44444444-4444-4444-4444-444444444444",
+            "55555555-5555-5555-5555-555555555555",
+            "66666666-6666-6666-6666-666666666666",
+            "77777777-7777-7777-7777-777777777777",
+            "88888888-8888-8888-8888-888888888888",
+            "99999999-9999-9999-9999-999999999999"})
+    public void testValidUuidStrings(final String uuidString) {
+        validate(uuidString);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaaa",
+            "aaaaaaaa+aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa+aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa+aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa+aaaaaaaaaaaa",
+            "`aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "{aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "@aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "[aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "/aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            ":aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "a:aaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aa:aaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaa:aaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaa:aaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaa:aa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaa:a-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-:aaa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-a:aa-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aa:a-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaa:-aaaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-:aaa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-a:aa-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aa:a-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaa:-aaaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-:aaa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-a:aa-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aa:a-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaa:-aaaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-:aaaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-a:aaaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aa:aaaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaa:aaaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaa:aaaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaa:aaaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaa:aaaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaa:aaaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa:aaa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaa:aa",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaa:a",
+            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa:"})
+    public void testInvalidUuidStrings(final String uuidString) {
+        try {
+            validate(uuidString);
```

Comment:
Use assertThrows ?

Reply:
Cool, forgot about that.  I also decided to use `assertDoesNotThrow` in the first test since otherwise there are no assertions.  That method makes the intent more obvious.